### PR TITLE
Bump uniffi to 0.26

### PR DESCRIPTION
**Pull request by reez**
### Description

Bump uniffi to 0.26 because the release adds support for things like: 
- async methods
- non_exhaustive enums

### Notes to the reviewers

versions of note

- `uniffi 0.26.1` (0.26.0 was yanked)
- `jna 5.14.0` [UniFFI relies on JNA for the ability to call native methods. JNA 5.12.0 or greater is required.](https://mozilla.github.io/uniffi-rs/kotlin/gradle.html#jna-dependency)
- `compileSdk = 33`

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### New Features:

* [ ] I've added tests for the new feature
* [ ] I've added docs for the new feature

#### Bugfixes:

* [ ] This pull request breaks the existing API
* [ ] I've added tests to reproduce the issue which are now passing
* [ ] I'm linking the issue being fixed by this PR

### CODE REVIEW

**Comment by reez** [comment]
> Could consider using the new `android_cleaner` flag for the Android build, see [lightningdevkit/ldk-node@3345ade](https://github.com/lightningdevkit/ldk-node/commit/3345aded7c32f0ae4bd0fe6310722d0e29b467a)

Thanks, you rule!

**Comment by reez** [comment]
(Just rebased on master to resolve new `cargo.lock` conflict)

File: bdk-android/lib/build.gradle.kts [hunk 1]
```diff
@@ -19,7 +19,7 @@ repositories {
 }
 
 android {
-    compileSdk = 31
+    compileSdk = 33
```
Comment (thunderbiscuit) [inline bdk-android/lib/build.gradle.kts:22]: Can we update the targetSdk to 31 as well?

File: bdk-android/lib/build.gradle.kts [hunk 2]
```diff
@@ -44,7 +44,7 @@ android {
 }
 
 dependencies {
-    implementation("net.java.dev.jna:jna:5.8.0@aar")
+    implementation("net.java.dev.jna:jna:5.12.0@aar")
```
Comment (thunderbiscuit) [inline bdk-android/lib/build.gradle.kts:47]: Small inconsistency between files, where some use 5.12 and some use 5.14.

File: .github/workflows/cont_integration.yml [hunk 3]
```diff
@@ -56,7 +56,7 @@ jobs:
         run: cargo clippy --all-targets --features "uniffi/bindgen-tests" -- -D warnings
 
       - name: "Test"
-        run: CLASSPATH=./tests/jna/jna-5.8.0.jar cargo test --features uniffi/bindgen-tests
+        run: CLASSPATH=./tests/jna/jna-5.14.0.jar cargo test --features uniffi/bindgen-tests
```
Comment (thunderbiscuit) [inline .github/workflows/cont_integration.yml:59]: See below comment on 5.14.

File: bdk-jvm/lib/build.gradle.kts [hunk 4]
```diff
@@ -59,7 +59,7 @@ tasks.withType<Test> {
 dependencies {
     implementation(platform("org.jetbrains.kotlin:kotlin-bom"))
     implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk7")
-    implementation("net.java.dev.jna:jna:5.8.0")
+    implementation("net.java.dev.jna:jna:5.12.0")
```
Comment (thunderbiscuit) [inline bdk-jvm/lib/build.gradle.kts:62]: I think this one should say 5.14.

**Review by thunderbiscuit — CHANGES_REQUESTED** [review]
Looks good to me. Just a few nits down below.

File: bdk-android/lib/build.gradle.kts (hunk 1 above)
Comment (reez) [reply bdk-android/lib/build.gradle.kts:22]: sure! updated, see line 26 for the `targetSdk = 33` change

File: bdk-android/lib/build.gradle.kts (hunk 2 above)
Comment (reez) [reply bdk-android/lib/build.gradle.kts]: thanks! changed.

File: bdk-jvm/lib/build.gradle.kts (hunk 4 above)
Comment (reez) [reply bdk-jvm/lib/build.gradle.kts]: changed to 5.14 now, thanks!

File: .github/workflows/cont_integration.yml (hunk 3 above)
Comment (reez) [reply .github/workflows/cont_integration.yml:59]: (line 62 is changed now)

**Review by thunderbiscuit — APPROVED** [review]
ACK 50f102bbd37ee514805ba70d71bfbbb6b0ae4bf5.

**Pull request merged**